dmoj/cli.py

```diff
@@ -310,6 +352,85 @@ def execute(self, line):
             problem, lang, src, tl, ml = data
             print ansi_style('#ansi[%s](yellow)/#ansi[%s](green) in %s' % (problem, i + 1, lang))
         print
+        
+
+class DifferenceCommand(Command):
+    name = 'diff'
+    help = 'Shows difference between two files.'
+
+    def _populate_parser(self):
+        self.arg_parser.add_argument('id_or_source_1', help='id or path of first source', metavar='<source 1>')
+        self.arg_parser.add_argument('id_or_source_2', help='id or path of second source', metavar='<source 2>')
+
+    def get_data(self, id_or_source):
+        err = None
+        src = None
+        try:
+            id = int(id_or_source)
+            global graded_submissions
+            problem, lang, src, tl, ml = graded_submissions[id - 1]
+        except IndexError:
```

Comment:
Put the `id = int(id_or_source)` into a different try block so you don't end up with some retarded control flow.